go-controller/pkg/ovn/egressgw.go

```diff
@@ -189,6 +189,22 @@ func (oc *Controller) addPodExternalGWForNamespace(namespace string, pod *kapi.P
 	if err != nil {
 		return fmt.Errorf("failed to ensure namespace locked: %v", err)
 	}
+	// build temporary ipTracker map to look for duplicate IPs
+	// key: IP, value: pod GW name
+	ipTracker := make(map[string]string)
```

Comment:
@trozet should this be a function that returns the sanitized list (and prints errors) since the same code is here and in `validateRoutingPodGWs()`?

Reply:
sure let me see if i can rework this to be a function that returns error if duplicate found. I think I will need to change how we pass around gateway infos, some places its a pointer